### ISSUE TITLE
Client-cert is not checking the cert validity time /FISH-5645

### ISSUE THREAD

**Issue by peterpz1** [payara/Payara#4763]
I have the endpoint defined to be authenticated using certificate. The cert is added to the truststore (cacerts.jks) and it is working, but it shouldn't be because cert is invalid (validity expired):
```2020-07-06 10:06:17.069 [http-thread-pool::http-listener-2(18)] DEBUG pl.orangelabs.pms.sdr.receiver.servlet.Fb3UploadServlet - cert=[
[
  Version: V3
  Subject: CN=sdmr-prod.itn.ftgroup, O=Orange, L=Lodz, C=PL
  Signature Algorithm: SHA256withRSA, OID = 1.2.840.113549.1.1.11

  Key:  Sun RSA public key, 2048 bits
  modulus: 25172384489365731032882167752978852808806809950077794566116719827434050089007516041968448028788314512438848135686922159235195540647272603007235478354162413707632870560751207134943967417145889283845108318003463017580119167377773510588392557413754484390388938390862185070764135111001449885915393381993725690540930595652755036013372415407331331783761752150648613588252984808672442649143711938840350493641530062903425118163386430664878018981564297856993000726704283877566017762447132201273596475971980835369576888355517395203482935169734662872257535312772036763510791615453548446276159886600869320093769315670484675839753
  public exponent: 65537
  Validity: [From: Tue Oct 10 14:53:39 CEST 2017,
               To: Thu Oct 10 14:53:39 CEST 2019]
```

Web.xml states:
```
<security-constraint>
        <web-resource-collection>
            <web-resource-name>Handle File From SDR</web-resource-name>
            <description></description>
            <url-pattern>/upload/*</url-pattern>
        </web-resource-collection>

        <auth-constraint>
            <description></description>
            <role-name>SDRLIGHT</role-name>
        </auth-constraint>
        <user-data-constraint>
            <transport-guarantee>CONFIDENTIAL</transport-guarantee>
        </user-data-constraint>
    </security-constraint>

    <login-config>
        <auth-method>CLIENT-CERT</auth-method>
    </login-config>
```

**Comment by peterpz1** [comment]
After manual calling the checkValidity method for the certificate retrieved from request we can confirm that it is not valid:
Log:
```
java.security.cert.CertificateExpiredException: NotAfter: Thu Oct 10 14:53:39 CEST 2019
	at sun.security.x509.CertificateValidity.valid(CertificateValidity.java:274)
	at sun.security.x509.X509CertImpl.checkValidity(X509CertImpl.java:629)
	at sun.security.x509.X509CertImpl.checkValidity(X509CertImpl.java:602)
```
Code:
```
X509Certificate[] certs = (X509Certificate[]) request.getAttribute("javax.servlet.request.X509Certificate");
            if (certs != null) {
                for (X509Certificate cert : certs) {
                    log.debug("cert=" + cert);
                    try {
                        cert.checkValidity();
                    } catch (Exception e) {
                        log.error("Validity", e);
                    }
                }
            }
```

**Comment by rdebusscher** [comment]
Hi,

The current Logic within Payara is to accept a certificate that is placed in the Trust Store of the domain. The validity of the Chain is checked, but not the certificate itself.

We are planning an extension so that more checks can be performed on the Certificate (like checking on CRL or OCSP, custom checks, etc ...) and also include the `checkValidity()` on the certificate.

This feature will be implemented via issue `FISH-387`.

Best Regards
Rudy

**Comment by smillidge** [comment]
Fixed by #5241

**Comment by lprimak** [comment]
Additional property of `validation-check` must be added to the client certificate realm for this check to be enabled

**Comment by lprimak** [comment]
Will be available in 2021.5 and after

**Comment by OndroMih** [comment]
The fix https://github.com/payara/Payara/pull/5241 was removed from the 5.2021.5 branch and we're going to implement this in a different way.

**Comment by rdebusscher** [comment]
This is fixed in 5.2021.8
